src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -298,4 +301,77 @@ private function getPrecisionFromCart(Cart $cart): int
 
         return $computingPrecision->getPrecision((int) $currency->precision);
     }
+
+    /**
+     * Update order details after a specific price has been created or updated
+     *
+     * @param Order $order
+     * @param OrderDetail $updatedOrderDetail
+     * @param Product $product
+     * @param int|null $combinationId
+     * @param int $productQuantity
+     * @param Number $priceTaxIncluded
+     * @param Number $priceTaxExcluded
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function updateOrderDetailsWithSameProduct(
+        Order $order,
+        OrderDetail $updatedOrderDetail,
+        Product $product,
```

Comment:
You don't need the Product nor the Combination as an input here The OrderDetail already has these informations through its fields `product_id` and `product_attribute_id`

Reply:
I think it's needed for comparison purpose, in order to check if the current orderDetail has the product we're looking for:

`
if ((int) $orderDetail->product_id !== (int) $product->id) {
   continue;
}
`